pkg/modeling/modeling.go

```diff
@@ -177,11 +162,11 @@ func (rs *ResourceSummary) AddToResourceSummary(crn ClusterResourceNode) {
 		found := false
 		// traverse linkedlist to add quantity of recourse modeling
 		for element := root.Front(); element != nil; element = element.Next() {
-			switch clusterResourceNodeComparator(element.Value, crn) {
+			switch rs.clusterResourceNodeComparator(element.Value, crn) {
 			case 0:
 				{
 					tmpCrn := element.Value.(ClusterResourceNode)
-					safeChangeNum(&tmpCrn.quantity, crn.quantity)
```

Comment:
@halfrost Can you remind me why we need the lock when adding the resource quantity to the `element`?

Reply:
+1

Reply:
This function `AddToResourceSummary` will be called concurrently. If no locking is performed, the eventual consistency of num will be incorrect.

Reply:
@jwcesign  I think the point where you're confused comes from whether it should be made a local variable. It is a global variable at first, so it needs to be locked here.

Reply:
Hi, @halfrost Thanks for your clarification!

So, this PR is reasonable and correct?

Reply:
Hi @jwcesign ,

Your changes are correct; there are no issues. With your modifications, each cluster now has its own modeling information.

There are no global variables left. In my previous PR, I didn't turn `defaultModelSorting` into a member variable because I was under the impression that we needed a broader global modeling context.

So this PR is prefect now.

Reply:
Thank you @halfrost for the clarification. 
I believe it reasonable for each cluster to have its resource models.

Reply:
OK, this pr makes sense. Thanks for @jwcesign work. 👍🏻